client/src/unifycr-sysio.c

```diff
@@ -706,12 +706,19 @@ off_t UNIFYCR_WRAP(lseek)(int fd, off_t offset, int whence)
 
 off64_t UNIFYCR_WRAP(lseek64)(int fd, off64_t offset, int whence)
 {
+    int tmpfd = fd;
+
     /* check whether we should intercept this file descriptor */
-    if (unifycr_intercept_fd(&fd)) {
-        /* ERROR: fn not yet supported */
-        fprintf(stderr, "Function not yet supported @ %s:%d\n", __FILE__, __LINE__);
-        errno = EBADF;
-        return (off64_t) (-1);
+    if (unifycr_intercept_fd(&tmpfd)) {
+        if (sizeof(off_t) == sizeof(off64_t))
+            return (off64_t) UNIFYCR_WRAP(lseek) (fd, (off_t) offset, whence);
+        else {
```

Comment:
This looks good to me.  I personally prefer braces around even single lines to prevent future bugs:

If (cond) {
  single line
} else {

However, I think our style checker script forces us to drop them for single-line blocks:

Reply:
Yes, I remember the last time when the checker complained when I put braces around a single statement.